sw/device/tests/dif/dif_otbn_sanitytest.c

```diff
@@ -0,0 +1,252 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * Information about an embedded OTBN application image.
+ *
+ * All pointers reference data in the normal CPU address space.
+ */
+typedef struct {
+  const char *imem_start;  //!< start of OTBN instruction memory
+  const char *imem_end;    //!< end of OTBN instruction memory
+  const char *dmem_start;  //!< start of OTBN data memory
+  const char *dmem_end;    //!< end of OTBN data memory
+} otbn_app_t;
+
+/**
+ * A function (entry point) in an OTBN application.
+ */
+typedef struct {
+  const otbn_app_t *app;  //!< application associated with this function
+  const char *func;       //!< address of the function entry point in normal CPU
+                          //!< address space
+} otbn_func_t;
+
+// Prepare info structs representing the barrett384 application, and the
+// wrap_barrett384() function. The goal of these structs is to make the linker-
+// generated symbols for the application image available in a form that is
+// easier to use.
+//
+// TODO: Think about macro-ifying or auto-generating some of this code in a
+// future OTBN driver.
+
+// See https://sourceware.org/binutils/docs/ld/Source-Code-Reference.html for
+// an in-depth explanation of how symbols are made available to C code.
```

Comment:
I don't think this comment is necessary. This is standard knowledge used throughout the OpenTitan codebase. Instead, I would link to the TU where these symbols are defined.

Reply:
I expanded the comment to explain where those symbols are from. 

I certainly don't feel that the details around accessing linker and assembly symbols is common knowledge (it wasn't for me). The docs even start with "Accessing a linker script defined variable from source code is not intuitive.", so I left it in.

Reply:
I filed https://github.com/lowRISC/opentitan/pull/4079 to add the "this is how OT does it" to the C/C++ style guide, so I think it's fair to get rid of the two line comment of general guidance. I hope this is a fair compromise.

Reply:
Thanks Sam. I removed the comment

Reply:
Yes, this is a good approach. In general, common patterns like this should go into the one-source-of-truth (the style guide).